### PR TITLE
improve backoff logic when writting to bigtable

### LINKED ISSUE #1209: increase bigtable backoff
When we will add multiple instances to our new bigtable-backed cluster, and write issues arise, having them all retry every 2s is too fast for bigtable to handle. + add jitter to randomize instead of having them retry in sync. we should make the retry configurable and default to a minute or so

see also #1099

### PR DESCRIPTION
**Pull request by woodsaj**
fixes #1209

### CODE REVIEW

**Comment by robert-milan** [comment]
We should wait to merge this until we come to a decision about interning later today.

File: idx/bigtable/bigtable.go [hunk 1]
```diff
@@ -394,6 +395,12 @@ func (b *BigtableIdx) processWriteQueue() {
 		pre := time.Now()
 		complete := false
 		attempts := 0
+		boff := &backoff.Backoff{
```
Comment (robert-milan) [inline idx/bigtable/bigtable.go]: Also, we can remove `attempts` now, as it is not used anywhere else.

Comment (Dieterbe) [reply idx/bigtable/bigtable.go]: or, do as store/bigtable does and regularly log a warning:
```
				if (attempts % 20) == 0 {
					log.Warnf("btStore: Failed to write %d chunks to bigtable. they will be retried. %s", len(rowKeys), err)
				}
```

note: we don't need our own var. can use `boff.Attempt()`

**Comment by Dieterbe** [comment]
seems unsafe to drop updates. isn't it better to keep retrying? that way it's clear when an operator needs to intervene and more obvious if a process restart will lead to an out-of-sync index.

the proposed implementation here can result in incorrect render responses if we don't include series because their lastUpdate hasn't been updated correctly. (we use the lastUpdate property from memory, but we may have restarted a fresh instance that loaded the index from the store and that hasn't seen data for that series so that lastUpdate field remains incorrect)

**Comment by woodsaj** [comment]
> seems unsafe to drop updates. 

It is a little, but it wont really have any impact.  The writes to bigtable here dont have a timeout, so the bigtable client library will continually retry if it can.  It will only return an error if the issue is un-recoverable, eg due to bigtable rejecting the write, rather then just failing to apply it.

Previously, if the write failed, we just ignored it. With this PR, we will now retry 3 times.  But to be honest, if the first one fails i am pretty certain all future attempts will also fail.  So looping forever until we succeed will just result in the write thread blocking on this write until someone restarts the instance.

I think a safer approach is to just rely on monitoring to alert when the idx.bigtable.query-insert.fail metric starts growing.

**Pull request merged**